src/pocket/PocketApiGatewayLambdaIntegration.ts

```diff
@@ -97,9 +105,79 @@ export class PocketApiGateway extends Resource {
         stageName: config.stage,
       }
     );
+    if (config.domain != null) {
+      this.createRoute53Record(config);
+    }
     this.addInvokePermissions();
   }
 
+  /**
+   * Sets up a custom domain name for the API gateway;
+   * adds ACM Certificate and configures Route 53 Record
+   * The deployed API stage will be exposed at the base path
+   * of the domain.
+   * e.g.
+   * abc123.execute-api.us-east-1.amazonaws.com/test-stage -> exampleapi.getpocket.dev
+   * @param config
+   */
+  private createRoute53Record(config: PocketApiGatewayProps) {
+    //Setup the Base DNS stack for our application which includes a hosted SubZone
+    const baseDNS = new ApplicationBaseDNS(this, `base-dns`, {
+      domain: config.domain,
+      tags: config.tags,
+    });
+
+    //Creates the Certificate for API gateway
+    const apiGatewayCertificate = new ApplicationCertificate(
+      this,
+      `api-gateway-certificate`,
+      {
+        zoneId: baseDNS.zoneId,
+        domain: config.domain,
+        tags: this.config.tags,
+      }
+    );
+
+    //setup custom domain name for API gateway
+    const customDomainName = new APIGateway.ApiGatewayDomainName(
+      this,
+      `api-gateway-domain-name`,
+      {
+        domainName: config.domain,
+        certificateArn: apiGatewayCertificate.arn,
+        dependsOn: [apiGatewayCertificate.certificateValidation],
+      }
+    );
+
+    new Route53.Route53Record(this, `apigateway-route53-domain-record`, {
+      name: customDomainName.domainName,
+      type: 'A',
+      zoneId: baseDNS.zoneId,
+      alias: [
+        {
+          evaluateTargetHealth: true,
+          name: customDomainName.cloudfrontDomainName,
+          zoneId: customDomainName.cloudfrontZoneId,
+        },
+      ],
+      dependsOn: [apiGatewayCertificate.certificateValidation],
+    });
+
+    new APIGateway.ApiGatewayBasePathMapping(
+      this,
+      `api-gateway-base-path-mapping`,
+      {
+        apiId: this.apiGatewayRestApi.id,
+        stageName: this.apiGatewayStage.stageName,
+        domainName: customDomainName.domainName,
+        basePath: config.basePath ?? '',
```

Comment:
Rather than a blank string, we should probably not include this argument if the input doesn't exist.

Reply:
i was thinking of writing something like this: 

```
if (config.basePath != null) { 
  //create ApiGatewayBasePathMapping with base path
else {
  //create ApiGatewayBasePathMapping without base path
}
````

but felt its getting slightly verbose so we can leave the basePath as empty string as it is not having any side effects.

Reply:
we checked this and the blank string doesn't not result in a path like `abc.getpocket.dev//endpoint`